aten/src/ATen/native/quantized/cpu/qnnpack/src/fc-prepack.cc

```diff
@@ -0,0 +1,53 @@
+#include <pytorch_qnnpack.h>
+#include <qnnpack/pack.h>
+#include <qnnpack_func.h>
+#include <cstring>
+
+namespace qnnpack {
+PackBMatrix::PackBMatrix(
+    size_t input_channels,
+    size_t output_channels,
+    uint8_t kernel_zero_point,
+    float kernel_scale,
+    const uint8_t* kernel,
+    const int32_t* bias) {
+  if (kernel_scale <= 0.0f || !std::isnormal(kernel_scale)) {
+    pytorch_qnnp_log_error(
+        "failed to create fully connected operator with %.7g kernel scale: "
+        "scale must be finite and positive",
+        kernel_scale);
+    assert("QNNPACK Runtime Error.");
+  }
+
+  const uint32_t nr = pytorch_qnnp_params.q8conv.nr;
+  const uint32_t kr = pytorch_qnnp_params.q8conv.kr;
+
+  const uint32_t n_stride = (output_channels + (nr - 1)) & -nr;
+  const uint32_t k_stride = (input_channels + (kr - 1)) & -kr;
+
+  input_channels_ = input_channels;
+  output_channels_ = output_channels;
+  packed_weights_ =
+      malloc(n_stride * (k_stride * sizeof(uint8_t) + sizeof(int32_t)));
```

Comment:
a) I would make this allocation aligned since this buffer is loaded-from in the micro kernels via SIMD intrinsics.  For this code to be correct, one needs to either a) check the implementations of all those kernels to make sure only unaligned load intrinsics are used everywhere for loads of packed_weights, or more simply b) allocate this buffer on a 16-byte boundary to guard against the case where aligned loads are used (which is actually preferable performance-wise but can trigger a processor exception on unaligned addresses depending on the architecture.)

If we support C++-17, check https://en.cppreference.com/w/cpp/memory/new/operator_new, otherwise https://en.cppreference.com/w/c/memory/aligned_alloc.

b) As an aside, n_stride and k_stride are already guaranteed to be powers of two themselves, but if aligned loads are actually used in the kernels, we still need to make sure each subsequent row starts on a 16-byte boundary (granted that the first one already does,) which means the number of bytes on each row of the matrix must be a multiple of 16 itself.

Reply:
Changed it to `aligned_alloc` with 16 byte alignment